### PR TITLE
serhii.trokhymets

### PR DESCRIPTION
**Pull request by SerhiiTrokhymets**
home work 3, task 1

### CODE REVIEW

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Easy_number
+
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: @SerhiiTrokhymets Hi! No need to add blank lines after all objects or lines with code. Read more[ here](https://github.com/rubocop-hq/ruby-style-guide#empty-lines-around-bodies)

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    end
+  end
+
+  private
+  attr_accessor :prime_arr, :bad_number
+  attr_reader :max_value
+
+end
+
+
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Excess blank line

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    (1..@prime_arr.size).each do
+    p prime_arr
+    prime_arr.rotate!
+    end
+  end
+
+  private
+  attr_accessor :prime_arr, :bad_number
+  attr_reader :max_value
+
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: [Excess blank line](https://github.com/rubocop-hq/ruby-style-guide#empty-lines-around-bodies)

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+end
+
+
+prime_matrix = Easy_number.new(10)
+prime_matrix.display_array
+
+
+
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: 2 excess blank line

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    find_prime_number
+    (1..@prime_arr.size).each do
+    p prime_arr
+    prime_arr.rotate!
+    end
+  end
+
+  private
+  attr_accessor :prime_arr, :bad_number
+  attr_reader :max_value
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: `attr_accessor` and `attr_reader` must be public

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
+class Easy_number
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Please, use [CamelCase for classes naming](https://github.com/rubocop-hq/ruby-style-guide#camelcase-classes)

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    find_prime_number
+    (1..@prime_arr.size).each do
+    p prime_arr
+    prime_arr.rotate!
+    end
+  end
+end
+
+prime_matrix = EasyNumber.new(10)
+prime_matrix.display_array
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Please, add an [empty line](https://thoughtbot.com/blog/no-newline-at-end-of-file) at the end of file.

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
+class EasyNumber
+  attr_accessor :prime_arr, :bad_number
+  attr_reader :max_value
+
+  def initialize (max_value)
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Do not put a space between a method name and the opening parenthesis

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    @max_value = max_value
+    @prime_arr = []
+    @bad_number = []
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Why initiate a variable that you do not use? The same is true for the attr_accessor argument

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def find_prime_number
+    (1..@max_value).each do |num|
+      if (1..num).select { |d| num % d == 0 } == [1, num]
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Favor modifier `if` usage when having a single-line body. Better style:
`prime_arr.push(num) if (1..num).select { |d| num % d == 0 } == [1, num]`

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        prime_arr.push(num)
+      end
+    end
+  end
+
+  def display_array
+    find_prime_number
+    (1..@prime_arr.size).each do
+    p prime_arr
+    prime_arr.rotate!
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Missed paddings

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    (1..@max_value).each do |num|
+      if (1..num).select { |d| num % d == 0 } == [1, num]
+        prime_arr.push(num)
+      end
+    end
+  end
+
+  def display_array
+    find_prime_number
+    (1..@prime_arr.size).each do
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Iterators know the number of array objects, so you do not need to pass them the exact figure
```suggestion
    @prime_arr.each do
      ...
```

**Review by hawkprovvv — APPROVED** [review]
Unresolved comments have been explained to the student, but he can't add corrections today.

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    p prime_arr
+    prime_arr.rotate!
+    end
+  end
+end
+
+prime_matrix = EasyNumber.new(10)
+prime_matrix.display_array
+
+
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: Excess empty line, must be only 1 empty line at the end of file

File: serhii.trokhymets/Home_work4/task1.rb [hunk 14]
```diff
@@ -0,0 +1,23 @@
+def dashatize (numbers)
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work4/task1.rb]: @SerhiiTrokhymets please, fix all annotations from the rubocop bot

File: serhii.trokhymets/Home_work 3/task_1.rb [hunk 15]
```diff
@@ -0,0 +1,26 @@
+class EasyNumber
```
Comment (hawkprovvv) [inline serhii.trokhymets/Home_work 3/task_1.rb]: @SerhiiTrokhymets please, [rename all directories](https://github.com/rubocop-hq/ruby-style-guide#snake-case-dirs)

File: serhii.trokhymets/homework_5/hm5_task1.rb [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+hash.each do |date, temp|
+  ttt = temp.inject(0) { |res, elem| res + elem } / temp.size
+  hash_month_aver[date] ||= ttt
+  hash_month_max[date] ||= temp.max
+  hash_month_min[date] ||= temp.min
+end
+
+hash_month_aver
+hash_month_max
+hash_month_min
```
Comment (hawkprovvv) [inline serhii.trokhymets/homework_5/hm5_task1.rb]: @SerhiiTrokhymets why are these methods called here?

Comment (SerhiiTrokhymets) [reply serhii.trokhymets/homework_5/hm5_task1.rb]: after debugging the code, I forgot to remove

**Pull request merged**